src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -377,17 +377,27 @@ impl<'a, P: PhysToVirt> MapperAllSizes for MappedPageTable<'a, P> {
             Err(PageTableWalkError::NotMapped) => return TranslateResult::PageNotMapped,
             Err(PageTableWalkError::MappedToHugePage) => {
                 let frame = PhysFrame::containing_address(p3[addr.p3_index()].addr());
+                let flags = p3[addr.p3_index()].flags();
                 let offset = addr.as_u64() & 0o_777_777_7777;
-                return TranslateResult::Frame1GiB { frame, offset };
+                return TranslateResult::Frame1GiB {
+                    frame,
+                    offset,
+                    flags,
+                };
```

Comment:
Here and elsewhere, if we're making this struct larger, it might make more sense to initialize it like:
```Rust
TranslateResult::Frame1GiB {
    frame: PhysFrame::containing_address(p3[addr.p3_index()].addr()),
    offset: addr.as_u64() & 0o_777_777_7777,
    flags:  p3[addr.p3_index()].flags(),
}
```

Reply:
I've changed the struct init accordingly, though I've left variables in two places, as it's defined in a match and I feel like that isn't as neat to look at.